[packages] Fix for cp2k and cosma

mtaillefumier opened the pull request:
- cosma: added C and fortran languages dependencies
- cp2k: All options in C2PK are off by default

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

Commit:
MPI verison of cp2k is not built at all

- cosma: added C and fortran languages dependencies
- cp2k: All options in C2PK are off by default

Signed-off-by: Mathieu Taillefumier <mathieu.taillefumier@free.fr>

Commit:
[package] Building unit tests can be turned off

RMeli commented on var/spack/repos/builtin/packages/cosma/package.py:
```diff
@@ -53,6 +53,8 @@ class Cosma(CMakePackage):
         variant("rccl", default=False, description="Use rocm rccl")
 
     depends_on("cxx", type="build")  # generated
```
```suggestion
    depends_on("cxx", type="build")
```

RMeli commented on var/spack/repos/builtin/packages/cp2k/package.py:
```diff
@@ -207,8 +215,8 @@ class Cp2k(MakefilePackage, CMakePackage, CudaPackage, ROCmPackage):
     depends_on("lapack")
-    depends_on("fftw-api@3")
-
+    depends_on("fftw-api@3", when="+fftw3")
```
The `fftw-api` is a virtual dependency. I think it should be always enabled, not just when one wants to use `fftw3` as the FFT library.

mtaillefumier replied:
OK, I will fix it asap

RMeli commented on var/spack/repos/builtin/packages/dbcsr/package.py:
```diff
@@ -162,12 +164,9 @@ def cmake_args(self):
             self.define_from_variant("BUILD_SHARED_LIBS", "shared"),
             self.define_from_variant("WITH_EXAMPLES", "examples"),
             self.define_from_variant("WITH_G2G", "g2g"),
+            self.define_from_variant("BUILD_TESTING", "test"),
```
```suggestion
            self.define_from_variant("BUILD_TESTING", "tests"),
```
?

mtaillefumier committed:
Update var/spack/repos/builtin/packages/cosma/package.py

Co-authored-by: Rocco Meli <r.meli@bluemail.ch>

Commit:
Fix typos

RMeli commented on var/spack/repos/builtin/packages/cp2k/package.py:
```diff
@@ -1050,6 +1057,8 @@ def cmake_args(self):
 
         args += [
+            self.define_from_variant("CP2K_USE_MPI", "mpi"),
+            self.define_from_variant("CP2K_USE_FFTW3", "fftw3"),
```
I think we should probably also deal with `CP2K_USE_FFTW3_WITH_MKL`, `CP2K_ENABLE_FFTW3_OPENMP_SUPPORT` and `CP2K_ENABLE_FFTW3_THREADS_SUPPORT` with the `fftw3` variant.

Commit:
Added missing variants

- force binutils >= 2.33 as binutils 2.30 has a bug that does not properly
  generate lib.so files.
- minor improvements in CP2K recipe.
